tests/MediaBrowser.Api.Tests/JellyfinApplicationFactory.cs

```diff
@@ -0,0 +1,124 @@
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+using Emby.Server.Implementations;
+using Emby.Server.Implementations.IO;
+using Emby.Server.Implementations.Networking;
+using Jellyfin.Drawing.Skia;
+using Jellyfin.Server;
+using MediaBrowser.Common;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.Mvc.Testing;
+using Microsoft.AspNetCore.TestHost;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+using Serilog;
+using Serilog.Extensions.Logging;
+
+namespace MediaBrowser.Api.Tests
+{
+    /// <summary>
+    /// Factory for bootstrapping the Jellyfin application in memory for functional end to end tests.
+    /// </summary>
+    public class JellyfinApplicationFactory : WebApplicationFactory<Startup>
+    {
+        private static readonly string _testPathRoot = Path.Combine(Path.GetTempPath(), "jellyfin-test-data");
+        private static readonly ConcurrentBag<ApplicationHost> _appHosts = new ConcurrentBag<ApplicationHost>();
```

Comment:
I wasn't actually sure if a thread-safe collection is necessary here, but I just threw it in to be safe. I think it might be necessary for running tests in parallel. Can switch to a regular collection if anyone can provide insight that shows it isn't needed.